### PR TITLE
[#5033] improvement(docs): Add the documents about access control

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?

Add the documents about access control.

### Why are the changes needed?

Fix: #5033

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Just docs.

### CODE REVIEW

File: docs/security/access-control.md [hunk 1]
```diff
@@ -507,6 +620,91 @@ boolean deleted =
...
+
+MetadataObject schema = ...
+Role role = client.revokePrivilegesFromRole("role1", schema, Lists.newArrayList(Privileges.SelectTable.allow()));
+
+MetadataObject table = ...
+Role role = client.revokePrivilegesFromRole("role1", table, Lists.newArrayList(Privileges.SelectTable.allow()));
+
+```
+
+
```
Comment (jerryshao) [inline docs/security/access-control.md:707]: Remove this one more blank line.

File: docs/security/access-control.md [hunk 2]
```diff
@@ -507,6 +620,91 @@ boolean deleted =
...
+GravitinoClient client = ...
+        
+MetadataObject schema = ...
+Role role = client.grantPrivilegesToRole("role1", schema, Lists.newArrayList(Privileges.SelectTable.allow()));        
+        
+MetadataObject table = ...
+Role role = client.grantPrivilegesToRole("role1", table, Lists.newArrayList(Privileges.SelectTable.allow()));
+```
+
+
```
Comment (jerryshao) [inline docs/security/access-control.md:664]: Remove one more blank line.

File: docs/security/access-control.md (hunk 1 above)
Comment (jerqi) [reply docs/security/access-control.md]: Removed.

File: docs/security/access-control.md (hunk 2 above)
Comment (jerqi) [reply docs/security/access-control.md]: Removed.

**Comment by jerqi** [comment]
I have tested by hand using OpenAPI and curl command.

File: docs/security/access-control.md [hunk 3]
```diff
@@ -481,6 +593,7 @@ Role role =
 </TabItem>
 </Tabs>
 
+
```
Comment (mchades) [inline docs/security/access-control.md:596]: unnecessary blank line?

File: docs/security/access-control.md [hunk 4]
```diff
@@ -507,6 +620,89 @@ boolean deleted =
...
+<TabItem value="java" label="Java">
+
+```java
+GravitinoClient client = ...
+        
+MetadataObject schema = ...
+Role role = client.grantPrivilegesToRole("role1", schema, Lists.newArrayList(Privileges.SelectTable.allow()));        
+        
+MetadataObject table = ...
+Role role = client.grantPrivilegesToRole("role1", table, Lists.newArrayList(Privileges.SelectTable.allow()));
```
Comment (mchades) [inline docs/security/access-control.md:661]: could you plz add comments to the demo code to make it more clearer and visual?

File: docs/security/access-control.md (hunk 3 above)
Comment (jerqi) [reply docs/security/access-control.md:596]: Removed.

File: docs/security/access-control.md (hunk 4 above)
Comment (jerqi) [reply docs/security/access-control.md:661]: Added.

File: docs/open-api/permissions.yaml [hunk 5]
```diff
@@ -207,23 +207,108 @@ paths:
...
+      summary: Grant privileges to a role
+      operationId: grantPrivilegeToRole
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/PrivilegeGrantRequest"
+            examples:
+              RoleRevokeRequest:
+                $ref: "#/components/examples/PrivilegeGrantRequest"
```
Comment (jerryshao) [inline docs/open-api/permissions.yaml:229]: “RoleRevokeRequest” I think it is not correct, right?

Comment (jerqi) [reply docs/open-api/permissions.yaml:229]: Corrected.

File: docs/security/access-control.md [hunk 6]
```diff
@@ -507,6 +619,93 @@ boolean deleted =
 
 ## Permission Operation
 
+### Grant privileges to a role
+
+You can revoke specific privileges from a role.
```
Comment (jerryshao) [inline docs/security/access-control.md:624]: "grant", not "revoke".

File: docs/security/access-control.md [hunk 7]
```diff
@@ -507,6 +619,93 @@ boolean deleted =
...
+
+```shell
+curl -X PUT -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" -d '{
+    "privileges": [
+      {
+      "name": "SELECT_TABLE",
+      "condition": "ALLOW"
+      }]
+}' http://localhost:8090/api/metalakes/test/permissions/roles/role1/schema/catalog1.schema1/revoke
```
Comment (jerryshao) [inline docs/security/access-control.md:638]: "grant" or "revoke"?

File: docs/security/access-control.md [hunk 8]
```diff
@@ -507,6 +619,93 @@ boolean deleted =
...
+
+```shell
+curl -X PUT -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" -d '{
+    "privileges": [
+      {
+      "name": "SELECT_TABLE",
+      "condition": "ALLOW"
+      }]
+}' http://localhost:8090/api/metalakes/test/permissions/roles/role1/schema/catalog1.schema1/grant
```
Comment (jerryshao) [inline docs/security/access-control.md:681]: Also here.

Comment (jerqi) [reply docs/security/access-control.md]: Corrected.

File: docs/security/access-control.md (hunk 7 above)
Comment (jerqi) [reply docs/security/access-control.md]: Changed to grant.

File: docs/security/access-control.md (hunk 6 above)
Comment (jerqi) [reply docs/security/access-control.md]: Changed to grant.

File: docs/security/access-control.md [hunk 9]
```diff
@@ -507,6 +619,93 @@ boolean deleted =
...
+GravitinoClient client = ...
+
+// Revoke the privilege allowing `SELEC_TABLE` for the `table` from `role1`         
+MetadataObject schema = ...
+Role role = client.revokePrivilegesFromRole("role1", schema, Lists.newArrayList(Privileges.SelectTable.allow()));
+
+// Revoke the privilege allowing `SELEC_TABLE` for the `table` from `role1`         
+MetadataObject table = ...
+Role role = client.revokePrivilegesFromRole("role1", table, Lists.newArrayList(Privileges.SelectTable.allow()));
+
```
Comment (yuqi1129) [inline docs/security/access-control.md:706]: You may miss some </TabItem> and </Tab> here.

Comment (jerqi) [reply docs/security/access-control.md:706]: I missed the comment.

**Comment by jerqi** [comment]
@jerryshao  I have checked this document of keyword `grant` and `revoke`.

File: docs/open-api/groups.yaml [hunk 10]
```diff
@@ -23,6 +23,42 @@ paths:
...
+    get:
+      tags:
+        - access control
+      summary: List groups (names)
+      operationId: listGroups
+      parameters:
+        - $ref: "#/components/parameters/details"
+      responses:
+        "200":
+          description: Returns the list of groups if {details} is true, otherwise returns the list of group name
```
Comment (yuqi1129) [inline docs/open-api/groups.yaml:35]: groups if -> group, if

File: docs/security/access-control.md [hunk 11]
```diff
@@ -152,9 +152,9 @@ You can also create a dedicated role for your business by API or the client.
 
 ### Permission privileges
 
-| Name          | Supports Securable Object | Operation              |
-|---------------|---------------------------|------------------------|
-| MANAGE_GRANTS | Metalake                  | grant or revoke a role |
+| Name          | Supports Securable Object | Operation                             |
+|---------------|---------------------------|---------------------------------------|
+| MANAGE_GRANTS | Metalake                  | grant or revoke a role or a privilege |
```
Comment (mchades) [inline docs/security/access-control.md:157]: grant or revoke a role?  It seems a bit strange. 

should it be `grant or revoke a privilege for a role` ?

Comment (jerqi) [reply docs/security/access-control.md:157]: Grant/Revoke a role for groups/users. Grant Revoke a privilege for roles.

Comment (mchades) [reply docs/security/access-control.md:157]: ok

Comment (jerqi) [reply docs/security/access-control.md]: I modified here.

File: docs/open-api/groups.yaml (hunk 10 above)
Comment (jerqi) [reply docs/open-api/groups.yaml:35]: I seems that we don't need the commar.

File: docs/security/access-control.md [hunk 12]
```diff
@@ -152,16 +152,16 @@ You can also create a dedicated role for your business by API or the client.
 
 ### Permission privileges
 
-| Name          | Supports Securable Object | Operation              |
-|---------------|---------------------------|------------------------|
-| MANAGE_GRANTS | Metalake                  | grant or revoke a role |
+| Name          | Supports Securable Object | Operation                                                                         |
+|---------------|---------------------------|-----------------------------------------------------------------------------------|
+| MANAGE_GRANTS | Metalake                  | Grant or revoke roles for a user or group. Grant or revoke privileges for a role. |
```
Comment (jerryshao) [inline docs/security/access-control.md:157]: Manages roles granted to or revoked from the user or group, and privileges granted to or revoked from the role.

File: docs/security/access-control.md [hunk 13]
```diff
@@ -507,6 +619,97 @@ boolean deleted =
...
+<TabItem value="java" label="Java">
+
+```java
+GravitinoClient client = ...
+
+// Revoke the privilege allowing `SELEC_TABLE` for the `table` from `role1`         
+MetadataObject schema = ...
+Role role = client.revokePrivilegesFromRole("role1", schema, Lists.newArrayList(Privileges.SelectTable.allow()));
+
+// Revoke the privilege allowing `SELEC_TABLE` for the `table` from `role1`
```
Comment (jerryshao) [inline docs/security/access-control.md:705]: The comment should be different.

File: docs/security/access-control.md (hunk 12 above)
Comment (jerqi) [reply docs/security/access-control.md]: Changed.

File: docs/security/access-control.md (hunk 13 above)
Comment (jerqi) [reply docs/security/access-control.md]: Changed from `table` to `schema`.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**